impexp-client-gui/src/main/java/org/citydb/gui/operation/common/filter/XMLQueryView.java

```diff
@@ -414,6 +431,32 @@ private String wrapQuery(String query) {
         return wrapper.toString();
     }
 
+    private void createSQLQuery() {
+        if (!ObjectRegistry.getInstance().getDatabaseController().connect()) {
+            return;
+        }
+
+        viewController.clearConsole();
+        log.info("Generating SQL query expression.");
+
+        AbstractDatabaseAdapter databaseAdapter = connectionPool.getActiveDatabaseAdapter();
+        SQLQueryBuilder sqlBuilder = new SQLQueryBuilder(
+                schemaMapping,
+                databaseAdapter,
+                BuildProperties.defaults().addProjectionColumn(MappingConstants.GMLID));
+
+        try {
+            QueryConfig queryConfig = unmarshalQuery();
+            ConfigQueryBuilder queryBuilder = new ConfigQueryBuilder(schemaMapping, databaseAdapter);
+            Query query = queryBuilder.buildQuery(queryConfig, ObjectRegistry.getInstance().getConfig().getNamespaceFilter());
+            Select select = sqlBuilder.buildQuery(query);
+            AbstractSQLAdapter sqlAdapter = databaseAdapter.getSQLAdapter();
+            log.print(SqlFormatter.format(select.toString(), sqlAdapter.getPlaceHolderValues(select)));
```

Comment:
Maybe we should better present the SQL query in a separate UI window instead of printing it to the console (similar to how we present generated Delete/Envelope scripts in the ADE manager)? We could use an RSyntaxArea then with SQL syntax highlighting. And there could be a "Save" button to the save SQL query to a file.

Reply:
see c485685